### ISSUE TITLE
Have screen reader acknowledge availability of search results

### ISSUE THREAD

**Issue by atsansone** [dart-lang/site-www#5793]
Hotlist: https://buganizer.corp.google.com/hotlists/5891206
Bug: https://buganizer.corp.google.com/issues/338010890

### Observed Results

The screen reader does not notify anything when the search results are available.

* [Screencast](https://screencast.googleplex.com/cast/NTQ2MTUzODg1MTc4MjY1Nnw1NGYzNzYwYi04YQ).

### Expected Results

The screen reader should notify when search results appear, for example "showing 10 search results".

### User Impact

Users will not know when results appear which may lead to confusion or lose context if the screen reader does not announce the search results when available. This will affect blind and low vision users who depend on screenreaders.

### Steps To Reproduce

1. Enable ChromeVox using <kbd>Ctrl</kbd> + <kbd>Alt</kbd> + <kbd>z</kbd>.
1. Go to https://dart.dev/guides.
1. Using the tab key navigate to "search" field.
1. Search for "isolate" and observe screen reader announcement when the search results appear.

### Relevent GAR Sheet

[A11Y Tracker](https://a11ytracker.corp.google.com/products/034ba691-ac78-4cc2-88c2-e06449cfd17e/edit/testresults?useCase=7jg80pq64n0p8&e=ProductLabels).

### GAR Details

[1.16C Final run through, with Screen reader](https://g3doc.corp.google.com/company/teams/gar/web/web-criteria.md?cl=head#run-through-screen-reader).

### Environment Details

* Machine: Chromebook.
* Browser: Chrome (123.0.6312.112 (Official Build) (64-bit)).
* Accessibility Tool: ChromeVox.
* Bug Filling Time: 10 mins.

**Comment by parlough** [comment]
These search results are generated, styled, and managed by the Google Custom Search setup we use.

Since we don't control that, I'm going to close this in favor of an umbrella issue https://github.com/dart-lang/site-www/issues/6486 which tracks implementing a new experience that we can properly integrate and style. Thanks!
